### PR TITLE
Add formula for bgpdump

### PR DESCRIPTION
**Pull request by dadrian**
This is a program maintained by RIPE for parsing BGP route data, and is
useful for network administrators and researchers.

### CODE REVIEW

File: Library/Formula/bgpdump.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+                          "--disable-dependency-tracking",
+                          "--disable-silent-rules",
+                          "--prefix=#{prefix}"
+    system "make"
+    system "make", "example" # Workaround for broken Makefile
+    system "make", "install"
+  end
+
+  test do
+    system "#{bin}/bgpdump"
```
Comment (DomT4) [inline Library/Formula/bgpdump.rb]: This isn't exiting cleanly. Can it be changed to do something a little more thorough? Alternatively you may wish to intercept the exit code to ensure it doesn't break.

File: Library/Formula/bgpdump.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  url "http://www.ris.ripe.net/source/bgpdump/libbgpdump-1.4.99.13.tgz"
+  sha256 "7551a285fa5c0885aec7290e1f316e58968baab70a4f3467f8f70868d338311e"
+
+  def install
+    system "./configure", "--disable-debug",
+                          "--disable-dependency-tracking",
+                          "--disable-silent-rules",
+                          "--prefix=#{prefix}"
+    system "make"
+    system "make", "example" # Workaround for broken Makefile
```
Comment (mikemcquaid) [inline Library/Formula/bgpdump.rb]: Please file an issue upstream about this and link it here, thanks!

Comment (dadrian) [inline Library/Formula/bgpdump.rb]: It looks like this has already been fixed upstream.

File: Library/Formula/bgpdump.rb (hunk 1 above)
Comment (dadrian) [inline Library/Formula/bgpdump.rb]: bgpdump has a -T flag that does a self-test. This test is broken on OS X in the most recent released version, however a patch is available upstream.

**Comment by dadrian** [comment]
@DomT4 and @mikemcquaid, it looks like both of these issues can be fixed by installing directly from the upstream Mercurial repository. It looks like it might just make more sense to install directly from the BitBucket repository at https://bitbucket.org/ripencc/bgpdump/overview. Thoughts?

**Comment by DomT4** [comment]
We don't really tap direct upstream repos outside of ` head ` and ` head-only ` unless there's submodules involved. Could we use the upstream commits that fixed the issues as patches? Something like:

```ruby
 
# Fix for issue x
patch do
  url "https://bitbucket.org/ripencc/bgpdump/commits/d16a154db1d5ae241423f8194bdeb5418eb59f61/raw/"
  sha256 "c02d544e938e188f79e67922beb1a240cbe21fc09eec5595027a75e3e0b844a1"
end
```

**Comment by mikemcquaid** [comment]
@dadrian We'll need to wait for a stable version before we include this.

**Comment by DomT4** [comment]
`1.4.99.14` was tagged at the end of last month. Is that stable/available for use here?

**Comment by dadrian** [comment]
I'm not sure, I'll take a look later today.

**Comment by DomT4** [comment]
Any idea if the upstream tag suggested is stable?

**Comment by dadrian** [comment]
@DomT4, sorry about the delay, this totally slipped off my plate.

It looks like the upstream is considerably more stable. It doesn't require the Makefile hack and the unit self-test `bgpdump -T` passes. However, the `make check` integration tests don't pass on my Yosemite install. I noted this upstream at https://bitbucket.org/ripencc/bgpdump/issue/20/testsh-fails-under-osx-gzip-issue

**Comment by DomT4** [comment]
Thanks for reporting that upstream @dadrian, Looks like they've said the fix will be available in the next version. They've [committed](https://bitbucket.org/ripencc/bgpdump/pull-request/8/fix-for-test-compared-compressed-output/diff) it already.

**Comment by dadrian** [comment]
@DomT4 They released another version upstream, and I updated the formula. Now all the tests pass (`make check` and `bgpdump -T`).

**Comment by mikemcquaid** [comment]
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

For future reference the preferred commit message format for new formulae is `bgpdump 1.4.99.15 (new formula)`.

**Pull request closed without merging**